src/mappings/entities/identities/mapPolyxTransaction.ts

```diff
@@ -132,14 +153,37 @@ export const handleTreasuryDisbursement = async (event: SubstrateEvent): Promise
 export const handleBalanceTransfer = async (event: SubstrateEvent): Promise<void> => {
   const args = extractArgs(event);
 
-  const [rawFromDid, rawFrom, rawToDid, rawTo, rawBalance, rawMemo] = args.params;
-
-  const amount = getBigIntValue(rawBalance);
-  const identityId = getTextValue(rawFromDid);
-  const address = getTextValue(rawFrom);
-  const toId = getTextValue(rawToDid);
-  const toAddress = getTextValue(rawTo);
-  const memo = bytesToString(rawMemo);
+  let address: string;
+  let toAddress: string;
+  let amount: bigint;
+  let memo: string | undefined;
+  let identityId: string | undefined;
+  let toId: string | undefined;
+
+  if (is8xChain(args.block)) {
+    const [rawFromAddress, rawToAddress, rawBalance] = args.params;
+    address = getTextValue(rawFromAddress);
+    toAddress = getTextValue(rawToAddress);
+    amount = getBigIntValue(rawBalance);
+    if (args.params.length > 3) {
+      const rawMemo = args.params[3];
+      memo = bytesToString(rawMemo);
+    }
+    identityId = (await Account.get(address))?.identityId;
+    toId = (await Account.get(toAddress))?.identityId;
+  } else {
+    const [rawFromDid, rawFrom, rawToDid, rawTo, rawBalance, rawMemo] = args.params;
+
+    amount = getBigIntValue(rawBalance);
+    identityId = getTextValue(rawFromDid);
+    address = getTextValue(rawFrom);
+    toId = getTextValue(rawToDid);
+    toAddress = getTextValue(rawTo);
+    if (args.params.length > 5) {
+      const rawMemo = args.params[5];
```

Comment:
Looks like this `rawMemo` shadows the outer `rawMemo`. Using a different name for the const here will make it clear about the intent. 

Otherwise this should at least get a comment for what spec version range we need this for.